File: algorithms/bitwise/python/AddTwoNumbersWithoutPlusOperator.py

# Python3 Program to add two numbers 
# without using arithmetic operator 
def Add(x, y): 
  
    # Iterate till there is no carry  
    while (y != 0): 
      
        # carry now contains common 
        # set bits of x and y 
        carry = x & y 
  
        # Sum of bits of x and y where at 
        # least one of the bits is not set 
        x = x ^ y 
  
        # Carry is shifted by one so that    
        # adding it to x gives the required sum 
        y = carry << 1
      
    return x 
  
print(Add(15, 32)) 

File: algorithms/bitwise/java/AddTwoNumbersWithoutPlusOperator.java

public class AddTwoNumbersWithoutPlusOperator {
    public static void main(String[] args) {
        int a = 10;
        int b = 90;
        add(a, b);
    }

    private static void add(int a, int b) {
        int carry;
        while (b != 0) {
            carry = a & b;
            a = a ^ b;
            b = carry << 1;
        }
        System.out.println("The sum  " + a);
    }
}
